src/config.ts

```diff
@@ -1,9 +1,27 @@
-import {Config, ConfigFn} from '../types/config'
+import {Config, ConfigFn, QueryAllElements, QueryElement} from '../types/config'
 import {prettyDOM} from './pretty-dom'
 
 type Callback<T> = () => T
+
+const queryElement: QueryElement = <T extends Element>(
+  element: T,
+  selector: string,
+) => element.querySelector(selector)
+const queryElementAll: QueryAllElements = <T extends Element>(
+  element: T,
+  selector: string,
+) => element.querySelectorAll(selector)
+
 interface InternalConfig extends Config {
   _disableExpensiveErrorDiagnostics: boolean
+  /**
+   * Returns the first element that is a descendant of node that matches selectors.
+   */
+  queryElement: QueryElement
```

Comment:
If this is not for public use (see declaration in `InternalConfig` not `Config`) let's prefix it with `_`. Otherwise move its declaration to `Config`